strategy/buyandhold/buyandhold.go

```diff
@@ -32,35 +47,44 @@ func (b *BuyAndHold) Run(
 	signals map[string]signal.Signal,
 ) strategy.StrategyOutput {
 	b.debug = map[string]string{}
+
+	// Every Nth day, rebalance to full diversification.
 	window := stores[dailywindow.Name].Get(date).Window
+	if b.days%b.period == 0 {
+		b.distribution = trading.NewBalancedCapitalDistribution(window.Open)
+	}
+
+	// Remove allocation to unlisted stocks.
+	allUnlistings := signals[unlistings.Name].Compute(date, stores).StringSet
+	for symbol, _ := range allUnlistings {
+		b.distribution.SetStock(symbol, 0.0)
+	}
 
-	if !b.initialized {
-		// Initialize by investing seed equally in all available stocks.
-		frac := b.desire.CashUSD / float64(len(window.Open))
-		for symbol, _ := range window.Open {
-			b.desire.Stocks[symbol] = frac / window.Close[symbol]
-		}
-		b.desire.CashUSD = 0.0
-		b.initialized = true
-	} else {
-		// For symbols that are no longer listed, divest at previous closing price.
-		for symbol, closePrice := range b.previousWindow.Close {
-			if _, ok := window.Open[symbol]; !ok {
-				b.desire.CashUSD += b.desire.Stocks[symbol] * closePrice
-				delete(b.desire.Stocks, symbol)
-			}
-		}
+	// Track performance.
+	perf := b.distribution.WindowPerformanceAtMeanPrice(window)
+	if b.days >= b.period {
+		b.periodPerf /= (1.0 + b.perf[b.days%b.period])
+	}
+	b.periodPerf *= (1.0 + perf)
+	b.perf[b.days%b.period] = perf
+	if perf > 0.0 {
+		b.wins++
+	} else if perf < 0.0 {
+		b.losses++
 	}
 
-	b.previousWindow = window
-	b.debug["value"] = fmt.Sprintf("%f", b.desire.Value(window.Close))
+	// Debug.
+	b.debug["periodperf"] = strconv.FormatFloat(b.periodPerf, 'G', -1, 64)
+	b.debug["perf"] = strconv.FormatFloat(perf, 'G', -1, 64)
+	b.debug["W/L"] = strconv.FormatFloat(float64(b.wins)/float64(b.losses), 'G', -1, 64)
 
-	return strategy.StrategyOutput{DesiredPortfolio: b.desire}
+	b.days++
+	return strategy.StrategyOutput{CapitalDistribution: &b.distribution}
```

Comment:
A question for us: Do we want to unit test strategies, or only test them empirically?

Reply:
TBD, resolving.